pygameMenu/menu.py

```diff
@@ -1332,26 +1305,38 @@ def _check_menu_initialized(self):
             raise Exception('The menu has not been initialized yet, try using mainloop function')
         return True
 
-    def mainloop(self, events=None, disable_loop=False):
+    def mainloop(self, surface, bgfun, events=None, disable_loop=False):
         """
         Main function of menu.
 
+        :param surface: Pygame surface to draw the menu
+        :type surface: pygame.surface.SurfaceType
+        :param bgfun: Background function called on each loop iteration before drawing the menu
+        :type bgfun: function
         :param events: Menu events
         :type events: list
         :param disable_loop: Disable infinite loop waiting for events
         :type disable_loop: bool
         :return: None
         """
+        assert isinstance(surface, _pygame.Surface)
+        assert callable(bgfun), 'background function must be callable (a function)'
+        assert isinstance(events, (list, type(None)))
+        assert isinstance(disable_loop, bool)
+
+        # Store the reference of the menu (if user moves through submenus)
         self._top = self
 
         if self.is_disabled():
             return
-        if self._actual._dopause and not disable_loop:
-            while True:
-                if self._main():
-                    return
-        else:
-            self._main(events)
+        while True:
+            bgfun()

```

Comment:
` bgfun()` should be called in the `draw()`method as it make a drawing ;-)

Reply:
> ` bgfun()` should be called in the `draw()`method as it make a drawing ;-)

but draw method does not require bgfun as a parameter. Should we add that?

Reply:
I tank you can keep the class attribute `self._background_function`. Set it in the `mainloop` method and use it in the `draw` method.

This permit to keep a consistence in actions performed by each method.

Reply:
That's better